modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java

```diff
@@ -251,6 +236,58 @@ private String _transform(String imgElementString, String src)
 		return replacement;
 	}
 
+	private void _transformImgTags(
+			String html, int start, int end, StringBundler sb)
+		throws PortalException {
+
+		int lastPos = start;
+
+		while (lastPos < end) {
+			int imgStart = html.indexOf(_OPEN_TAG_TOKEN_IMG, lastPos);
+
+			if ((imgStart == -1) || (imgStart > end)) {
+				sb.append(html.substring(lastPos, end));
+
+				return;
+			}
+
+			sb.append(html.substring(start, imgStart));
+
+			int imgEnd = html.indexOf(CharPool.GREATER_THAN, imgStart) + 1;
+
+			int attributeListPos = imgStart + _OPEN_TAG_TOKEN_IMG.length();
+
+			int srcStart = html.indexOf(_ATTRIBUTE_TOKEN_SRC, attributeListPos);
+
+			if (srcStart == -1) {
+				sb.append(html.substring(imgStart, imgEnd));
+
+				lastPos = imgEnd;
+
+				continue;
+			}
+
+			int quotePos = srcStart + _ATTRIBUTE_TOKEN_SRC.length();
```

Comment:
maybe _ATTRIBUTE_TOKEN_SRC.length() should take the end on "?" if exists?

Reply:
I've updated the regexp to accept optional query parameters. Jsoup was covering a bug in the regexp.